lib/std/heap/general_purpose_allocator.zig

```diff
@@ -167,13 +175,14 @@ pub fn GeneralPurposeAllocator(comptime config: Config) type {
         total_requested_bytes: @TypeOf(total_requested_bytes_init) = total_requested_bytes_init,
         requested_memory_limit: @TypeOf(requested_memory_limit_init) = requested_memory_limit_init,
 
-        mutex: @TypeOf(mutex_init) = mutex_init,
+        mutex: MutexType = if (config.MutexType) |_| undefined else mutex_init,
```

Comment:
I support making GPA configurable in this way. However, can you move this logic to `mutex_init` and keep the previous code the same?

Also shouldn't it be:

```zig
if (config.MutexType) |T| T{}
```

instead of `undefined`?

Reply:
Will this cause missing field error if the parent type T has a required field?

Reply:
I think I push the logic to mutex_init relatively easily.

Reply:
> Will this cause missing field error if the type T has a required field?

Yes - but that's better than initializing it with undefined right? Essentially we are defining the duck typing interface here, and I think for a mutex it's reasonable to expect it to be initializable with no explicit fields.

Reply:
I see your point.  an implementer of T can always assign their field as "undefined" as its initial value.